explauto/interest_model/riac.py

```diff
@@ -0,0 +1,947 @@
+import time
+import numpy as np
+import matplotlib.pyplot as plt
+from heapq import heappop, heappush
+from scipy.spatial.kdtree import minkowski_distance_p
+
+from .interest_model import InterestModel
+from .competences import competence_exp, competence_dist
+from ..utils.utils import rand_bounds
+from ..utils.config import make_configuration
+
+
+
+class RiacInterest(InterestModel):
+    def __init__(self, 
+                 conf, 
+                 expl_dims, 
+                 max_points_per_region, 
+                 split_mode, 
+                 competence_measure, 
+                 progress_win_size, 
+                 progress_measure, 
+                 sampling_mode):
+        
+        self.conf = conf
+        self.bounds = self.conf.bounds[:, expl_dims]
+        self.competence_measure = competence_measure
+        
+        if progress_win_size >= max_points_per_region:
+            print "WARNING: progress_win_size should be < max_points_per_region"
+        
+        self.data_x = None
+        self.data_c = None
+
+        self.tree = Tree(self.get_data_x, 
+                         np.array(self.bounds, dtype=np.float), 
+                         self.get_data_c, 
+                         max_points_per_region=max_points_per_region, 
+                         split_mode=split_mode, 
+                         progress_win_size=progress_win_size, 
+                         progress_measure=progress_measure, 
+                         sampling_mode=sampling_mode,
+                         idxs=[])
+        
+        InterestModel.__init__(self, expl_dims)
+
+    def sample(self):
+        return self.tree.sample()
+    
+    def progress(self):
+        return self.tree.progress
+    
+    def get_data_x(self):
+        return self.data_x
+    
+    def get_data_c(self):
+        return self.data_c
+    
+    def update(self, xy, ms):
+        #print np.shape(self.data_x), np.shape(np.array([xy[self.expl_dims]]))
+        if self.data_x is None:
+            self.data_x = np.array([xy[self.expl_dims]])
+        else:
+            self.data_x = np.append(self.data_x, np.array([xy[self.expl_dims]]), axis=0)
+        if self.data_c is None:
+            self.data_c = np.array([self.competence_measure(xy, ms)])
+        else:
+            self.data_c = np.append(self.data_c, self.competence_measure(xy, ms)) 
+        self.tree.add(np.shape(self.data_x)[0] - 1)
+
+
+
+
+
+# Based on scipy.spatial.KDTree
```

Comment:
I'm not quite sure, but I'm thinking that instead of copy/pasting the code from the scipy KDTree it should be feasible to use inheritance (or delegation?) instead.